tendermint/tests/lite.rs

```diff
@@ -125,6 +205,70 @@ fn run_test_cases(cases: TestCases) {
     }
 }
 
+// Link to the commit where the happy_path.json was created:
+// https://github.com/Shivani912/tendermint/commit/89aa17ab9ae0a76941eb15b7957452ec78ce5696
+#[test]
+fn bisection_simple() {
+    let case: TestBisection =
+        serde_json::from_str(&read_json_fixture("many_header_bisection/happy_path")).unwrap();
+    run_bisection_test(case);
+}
+
+fn run_bisection_test(case: TestBisection) {
+    println!("{}", case.description);
+
+    let untrusted_height = case.height_to_verify.try_into().unwrap();
+    let trust_threshold = case.trust_level;
+    let trusting_period = case.trust_options.period;
+    let now = case.now;
+
+    let provider = case.primary;
+    let req = MockRequester::new(provider.chain_id, provider.lite_blocks);
+
+    let expects_err = match &case.expected_output {
+        Some(eo) => eo.eq("error"),
+        None => false,
+    };
+
+    let trusted_height = case.trust_options.height.try_into().unwrap();
+    let trusted_header = &req
+        .signed_header(trusted_height)
+        .expect("could not 'request' signed header");
+    let trusted_vals = &req
+        .validator_set(trusted_height + 1)
+        .expect("could not 'request' validator set");
+
+    let trusted_state = TrustedState::new(trusted_header, trusted_vals);
+
+    match lite::verify_bisection(
+        trusted_state,
+        untrusted_height,
+        trust_threshold,
+        &trusting_period.into(),
+        &now.into(),
+        &req,
+    ) {
+        Ok(new_states) => {
+            let untrusted_signed_header = &req
+                .signed_header(untrusted_height)
+                .expect("header at untrusted height not found");
+            let untrusted_next_vals = &req
+                .validator_set(untrusted_height + 1)
+                .expect("val set at untrusted height not found");
+            let expected_state = TrustedState::new(
+                &untrusted_signed_header.to_owned(), //.into(),
```

Comment:
What's with the comment here? Looks like this is obsolete now?
```suggestion
                &untrusted_signed_header.to_owned(),
```